### DISCUSSION TITLE
Dual layers electrodes

### DISCUSSION THREAD

**Discussion by MarcAntoine88** [pybamm-team/PyBaMM#2024] [Q&A]
Hello PyBaMM Community, I would like to see the effects of dual layers electrodes (can be anode or cathode) on the cell behavior and I'm wondering how hard it would be to implement it in PyBaMM? The idea is to superimpose 2 electrodes with different properties (particle size, etc) to get the cathode for instance.  This would require for example 2 domains for the positive electrode. Do you think PyBaMM is flexible enough to accommodate this? Thanks for the help!

**Comment by valentinsulzer** [comment]
Can you provide a complete list of the parameters that need to change between the two electrodes? If it's just particle size, volume fraction, etc it's easy, otherwise it might be more complicated

**Reply by MarcAntoine88** [reply]
Thanks @tinosulzer, if I take it the other way around, what would be a complicated parameter to deal with ? I would like to try let's say with two different cathodes so that would be particle size, porosity, tortuosity, diffusivity & reaction rate. What would be the best strategy to tackle it according to your experience with PyBaMM ?

**Reply by valentinsulzer** [reply]
> particle size, porosity

These can already be given as a function of `x`, so you can use the heaviside function. You just need to make sure you define your mesh in such a way that the boundary between the two electrodes falls exactly at an edge in the mesh (e.g. use enough discretization points to ensure this, or define a custom mesh)

> diffusivity & reaction rate

these require editing the source code so that they can be given as a function of `x`

> tortuosity

are you using Bruggeman or specifying tortuosity manually

**Reply by MarcAntoine88** [reply]
> > particle size, porosity
> 
> Yes you're right I didn't think about it but using the heaviside function would be practical! 
> 
> > diffusivity & reaction rate
> 
> ok I'll look at it :) 
> 
> > tortuosity
> 
> I'm using Bruggeman for now but if I need to change to manual tortuosity it's possible as well

Thanks @tinosulzer

**Reply by dion-w** [reply]
I already did the mentioned 2 layer setup in PyBaMM and setting it up with function parameters and Heaviside functions works like a charm.

**Reply by MarcAntoine88** [reply]
> I already did the mentioned 2 layer setup in PyBaMM and setting it up with function parameters and Heaviside functions works like a charm.

Hi @dion-w I've done it for particle size last week and it works well indeed!

**Reply by anmolpan** [reply]
Hi @MarcAntoine88 , could you share your code with dual particle sizes? 
Thank you!
